src/editor/object_option.hpp

```diff
@@ -68,12 +70,31 @@ class ObjectOption
   const std::string m_key;
   const unsigned int m_flags;
 
+private:
+  BaseObjectOption(const BaseObjectOption&) = delete;
+  BaseObjectOption& operator=(const BaseObjectOption&) = delete;
+};
+
+template <typename T = void>
+class ObjectOption: public BaseObjectOption
+{
+public:
+  ObjectOption(const std::string& text, const std::string& key, unsigned int flags, T* pointer = nullptr);
+  virtual ~ObjectOption() override
+  {
+  }
+
+  virtual T* get_value() const { return m_value_pointer; }
+
+protected:
+  T* const m_value_pointer;
+
 private:
   ObjectOption(const ObjectOption&) = delete;
   ObjectOption& operator=(const ObjectOption&) = delete;
 };
 
-class BoolObjectOption : public ObjectOption
+class BoolObjectOption : public ObjectOption<bool>
```

Comment:
Just a suggestion, but since this PR edits all object options, `final` could be added to the class declarations.